### PR TITLE
[INFRA-2243] Switch to new buildPluginWithGradle() step in Jenkinsfile

### PR DESCRIPTION
**Pull request by darxriggs**
### [INFRA-2243](https://issues.jenkins-ci.org/browse/INFRA-2243) Split Maven and Gradle steps in Pipeline Library

There should be no difference in behaviour at the moment.
The existing `buildPlugin()` step is deprecated for Gradle-based plugins.
The new step is planned to get new Gradle/Groovy functionality over time.

Skipping tests as the pipeline on ci.jenkins.io otherwise tries to collect
test results which would fail because no tests have been implemented.

### CODE REVIEW

**Comment by darxriggs** [comment]
This change is first taken into account when merged. This is because I have no write access to the repository. See the relevant output from the Jenkins log:
> [2020-01-02T22:23:49.506Z] ‘Jenkinsfile’ has been modified in an untrusted revision

**Review by oleg-nenashev — APPROVED** [review]
Fine with me. It should not change the behavior

**Comment by darxriggs** [comment]
@rodrigc can you merge this? We want to migrate all Gradle based plugins to the new step.

**Comment by darxriggs** [comment]
@rodrigc friendly reminder

**Comment by darxriggs** [comment]
@rodrigc another friendly reminder

**Comment by jimklimov** [comment]
Seems @rodrigc is not longer responding, so I took up maintenance of the plugin. Recent changes proposed in #26 also included `buildPluginWithGradle` although that PR supplied a dummy test file for similar effect (have something to run and analyze). The 0.7 release included that change.

I am a bit at a loss here which path is better: not-running tests is less resource intensive than running and analyzing a dummy; but having a dummy might nudge someone to actually code some tests - and pipeline is ready to handle them (without a change to master Jenkinsfile, like what blocked this PR earlier). I suppose the resource intensiveness difference is marginal, so a chance to get someone to make real tests wins?.. :)

**Pull request closed without merging**
